pkg/policies/engine/rego/rego.go

```diff
@@ -113,6 +138,40 @@ func (r *Rego) Verify(ctx context.Context, policy *engine.Policy, input []byte,
 	return violations, nil
 }
 
+// loadEnvironmentCapabilities loads the capabilities of the environment
+// based on the mode of operation, defaulting to EnvironmentModeRestrictive if not provided.
+func loadEnvironmentCapabilities(mode EnvironmentMode) *ast.Capabilities {
+	capabilities := ast.CapabilitiesForThisVersion()
+	var builtins []*ast.Builtin
+
+	switch mode {
+	case EnvironmentModeRestrictive:
```

Comment:
can you encapsulate this behaviour as part of the `EnvironmentMode` type? Something like:
```
func (emr EnvironmentModeRestrictive) Capabilities() *ast.Capabilities {
  localBultins := make ...
 ...
}

func (rmp EnviromentModePermissive) Capabilities() *ast.Capabilities {
  return ast.CapabilitiesForThisVersion()
}
```
This way, the change above would look like this:
```
res, err := queryRego(ctx, mainRule, parsedModule, regoInput, regoFunc, rego.Capabilities(r.OperatingMode.Capabilities()))
```